cmd/batch-job-common-types.go

```diff
@@ -81,3 +83,52 @@ func (r BatchJobRetry) Validate() error {
 
 	return nil
 }
+
+// BatchJobSizeFilter supports size based filters - LesserThan and GreaterThan
+type BatchJobSizeFilter struct {
+	UpperBound BatchJobSize `yaml:"lesserThan,omitempty" json:"lesserThan"`
+	LowerBound BatchJobSize `yaml:"greaterThan,omitempty" json:"greaterThan"`
```

Comment:
@shtripat I have renamed the internal fields so that the methods on them are easy to understand. I have also added unit tests to `BatchJobSizeFilter` methods to cover typical scenarios and to illustrate its usage.